core/src/main/java/org/pentaho/platform/dataaccess/datasource/ui/selectdialog/DatasourceSelectionDialogController.java

```diff
@@ -469,6 +473,12 @@ public void onDialogError( String value ) {
 
   @Bindable
   public void removeDatasourceConfirm() {
+    if ( messageBundle != null ) {
+      XulLabel removeDatasourceConfirmationDialogLabel = (XulLabel) removeDatasourceConfirmationDialog.getElementById( "removeDatasourceConfirmationDialogLabel" );
+      LogicalModelSummary logicalModelSummary = getDialogResult();
```

Comment:
This method happily returns null from time to time. Kaboom? And yeah, I know, that binding probably should not be active if there is no selection. But at the same time would it hurt to add a explicit 

     if (datasourceSelectionDialogModel.getSelectedIndex() == -1) return; 

to the method to protect it?